pkg/httpd/simpleserver.go

```diff
@@ -48,12 +50,31 @@ func (simpleServer *SimpleServer) Start() error {
 	mux.HandleFunc("/healthz", webhook.HealthCheckHandler)
 	mux.HandleFunc("/mutate", admissionHandler.HandleAdmission)
 
+	if simpleServer.MetricsPort != nil && *simpleServer.MetricsPort != simpleServer.Port {
+		go startMetricsServer(fmt.Sprintf(":%d", *simpleServer.MetricsPort))
+	} else {
+		mux.Handle("/metrics", promhttp.Handler())
+	}
+
 	if simpleServer.Local {
 		return server.ListenAndServe()
 	}
 	return server.ListenAndServeTLS(simpleServer.CertFile, simpleServer.KeyFile)
 }
 
+func startMetricsServer(addr string) {
```

Comment:
Change this to inherit simpleServer and re-use the defined prom handler so its consistent across servers.

```
func (simpleServer *SimpleServer) startMetricsServer(metricsHandler http.Handler) {
        ...
	metricsRouter.Handle("/metrics", metricsHandler)

	metricsServer := &http.Server{
		Addr:    fmt.Sprintf(":%d", simpleServer.MetricsPort),
		Handler: metricsRouter,
	}
       ....
}
```

```
metricsHandler := promhttp.Handler()

if simpleServer.MetricsPort != simpleServer.Port {
	go simpleServer.startMetricsServer(metricsHandler)
} else {
	mux.Handle("/metrics", metricsHandler)
}

Reply:
done